app/authentication/jti_claim_storage.py

```diff
@@ -26,26 +26,29 @@ def _is_valid(jti_claim):
     return True
 
 
-def use_jti_claim(jti_claim):
-    """
-    Use a jti claim
-    :param jti_claim: jti claim to mark as used.
-    :raises ValueError: when jti_claim is None.
-    :raises TypeError: when jti_claim is not a valid uuid4.
-    :raises JtiTokenUsed: when jti_claim has already been used.
-    """
-    if jti_claim is None:
-        raise ValueError
-    if not _is_valid(jti_claim):
-        logger.info("jti claim is invalid", jti_claim=jti_claim)
-        raise TypeError
+class JtiClaimStorage:
 
-    try:
-        with commit_or_rollback(db_session):
+    def __init__(self, database):
+        self._database = database
+
+    def use_jti_claim(self, jti_claim):
+        """
+        Use a jti claim
+        :param jti_claim: jti claim to mark as used.
+        :raises ValueError: when jti_claim is None.
+        :raises TypeError: when jti_claim is not a valid uuid4.
+        :raises JtiTokenUsed: when jti_claim has already been used.
+        """
+        if jti_claim is None:
+            raise ValueError
+        if not _is_valid(jti_claim):
+            logger.info("jti claim is invalid", jti_claim=jti_claim)
+            raise TypeError
+
+        try:
             jti = UsedJtiClaim(jti_claim)
-            # pylint: disable=maybe-no-member
-            # db_session has an add function but it is wrapped in a session_scope which confuses pylint
-            db_session.add(jti)
-    except IntegrityError as e:
-        logger.error("jti claim has already been used", jti_claim=jti_claim)
-        raise JtiTokenUsed(jti_claim) from e
+            self._database.db_session.add(jti)
```

Comment:
Just wonder rather than accessing `db_session` whether it would be cleaner to make the `_db_session` a private variable and have an `add_and_commit(entity)` method on `Database` and let Database worry about talking to the session?